easysocial/libraries/mappingHelper.php

```diff
@@ -1290,6 +1277,140 @@ public function groupSchema($rows = null, $userid = 0)
 		return $result;
 	}
 
+	// Function for create page schema
+	/**
+	 * To build ablum object
+	 *
+	 * @param   string  $rows    array of data
+	 * @param   int     $userid  user id
+	 * 
+	 * @return array
+	 *
+	 * @since 2.0
+	 */
+	public function pageSchema($rows = null, $userid = 0)
+	{
+		if ($rows == null || $userid == 0)
+		{
+			return null;
+		}
+
+		$result  = array();
+		$user   = FD::user($userid);
+
+		// Easysocial default profile
+		$profile = $user->getProfile();
+		$fmod_obj = new EasySocialModelFields;
+
+		foreach ($rows as $ky => $row)
+		{
+			$page = ES::page($row->id);
+			$page_model		=	FD::model('Pages');
+			$steps = $page_model->getAbout($page, $activeStep);
+
+			$fieldsArray = array();
+
+			// Get custom fields model.
+			$fieldsModel = FD::model('Fields');
+
+			// Get custom fields library.
+			$fields      = FD::fields();
+			$field_arr   = array();
+
+			if ($steps)
+			{
+				$pageInfo = new stdClass;
+
+				foreach ($steps as $step)
+				{
+					foreach ($step->fields as $pageInfo)
+					{
+						$fobj = new fildsSimpleSchema;
+						$fobj->field_id = $pageInfo->id;
+						$fobj->unique_key = $pageInfo->unique_key;
+						$fobj->title = JText::_($pageInfo->title);
+						$fobj->field_name = JText::_($pageInfo->title);
+						$fobj->step = $pageInfo->step_id;
+						$fobj->field_value = $pageInfo->output;
+
+						if ($fobj->unique_key == 'DESCRIPTION')
+						{
+							$fobj->field_value = preg_replace("~[[:alpha:]]+://[^<>[:space:]]+[[:alnum:]/]~", "<a href=\"\\0\">\\0</a>", $fobj->field_value);
+						}
+
+						array_push($fieldsArray, $fobj);
+					}
+				}
+			}
+
+			if (isset($row->id))
+			{
+				$pageobj = FD::page($row->id);
+				$item   = new PageSimpleSchema;
+				$item->id           = $row->id;
+				$item->type = 'page';
+				$item->title        = $row->title;
+				$item->alias        = $row->alias;
+				$item->description  = $row->description;
+				$item->hits         = $row->hits;
+				$item->state        = $row->state;
+				$item->created_date = $this->dateCreate($row->created);
+
+				// Get category name
+				$category = FD::table('PageCategory');
+				$category->load($row->category_id);
+				$item->category_id   = $row->category_id;
+				$item->category_name = $category->get('title');
+				$item->page_type = $row->type;
+				$item->cover         = $pageobj->getCover();
+				$pageobj               = FD::page($row->id);
+				$x                    = $pageobj->cover->x;
+				$y                    = $pageobj->cover->y;
+				$item->cover_position = $x . '% ' . $y . '%';
+				$item->created_by   = $row->creator_uid;
+				$item->creator_name = JFactory::getUser($row->creator_uid)->username;
+				$item->params       = (!empty($row->params)) ? $row->params : false;
+				$item->more_info = $fieldsArray;
+
+				foreach ($row->avatars As $ky => $avt)
+				{
+					$avt_key        = 'avatar_' . $ky;
+					$item->$avt_key = JURI::root() . 'media/com_easysocial/avatars/page/' . $row->id . '/' . $avt;
```

Comment:
@Pratikshakhandagale - What will be the path if I am S3. Check this

Reply:
@snehal0904 - I will send new MR to getting s3 image path